### PR TITLE
AzureMonitor: Refactor datasource.query to fix --strict error

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Refactors the AzureMonitor datasource query function to improve the typescript and remove one --strict error.

`Object.key(...)` will always return `string[]`, so I moved to a Map preserves the types we want

**Which issue(s) this PR fixes**:

Related to #32142

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts [hunk 1]
```diff
@@ -84,28 +68,28 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
...
+      if (!byType.has(target.queryType)) {
+        const queryForType = _.cloneDeep(options);
+        queryForType.requestId = `${queryForType.requestId}-${target.refId}`;
+        queryForType.targets = [];
+        byType.set(target.queryType, queryForType);
       }
-      q.targets.push(target);
+
+      const queryForType = byType.get(target.queryType);
+      queryForType && queryForType.targets.push(target);
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:79]: nit: can be simplified:
```suggestion
      queryForType?.targets.push(target);
```

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts [hunk 2]
```diff
@@ -290,3 +274,20 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     return this.templateSrv.getVariables().map((v) => `$${v.name}`);
   }
 }
+
+function migrateQuery(target: AzureMonitorQuery) {
+  if (target.queryType === AzureQueryType.ApplicationInsights) {
+    if ((target.appInsights as any).rawQuery) {
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:280]: This type cast seems error prone

**Review by hugohaggmark — COMMENTED** [review]
Looks good to me, haven't tested it though.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts (hunk 2 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:280]: Yes 100% true. I actually have no idea what this all does (apart from 'migrate') - i just moved it out into a seperate function. I'll take a stab at it and see if i can clean it up.

**Review by sunker — APPROVED** [review]
Looks good. Tested for metrics and it works fine.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts (hunk 2 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:280]: I tried to take a stab at improving this, but this seems wrong/I don't understand it, so I'm scared that making this typesafe will actually regress. 

I'm going to leave this as-is for now until I can get some guidance on what this code is supposed to actually do.

**Pull request merged**